packages/core/strapi/src/cli/commands/transfer/command.ts

```diff
@@ -44,14 +44,128 @@ const command = () => {
+                  default: process.env.STRAPI_TRANSFER_URL,
+                  validate(input) {
+                    try {
+                      const url = new URL(input);
+                      if (!['http:', 'https:'].includes(url.protocol)) {
+                        return 'URL must use http: or https: protocol';
+                      }
+                      return true;
+                    } catch (error) {
+                      return 'Please enter a valid URL (e.g., http://localhost:1337 or https://example.com)';
+                    }
+                  }
+                },
+                {
+                  type: 'password',
+                  name: 'token',
+                  message: 'Enter your transfer token for the remote Strapi source:',
+                  default: process.env.STRAPI_TRANSFER_TOKEN,
+                  validate(input) {
+                    if (!input?.length) {
+                      return 'Transfer token is required';
+                    }
+                    return true;
+                  }
+                }
+              ]);
+              opts.from = new URL(remoteUrl);
+              opts.fromToken = token;
+            } else {
+              const { remoteUrl, token } = await inquirer.prompt([
+                {
+                  type: 'input',
+                  name: 'remoteUrl',
+                  message: 'Enter the URL of the remote Strapi instance to send data to:',
+                  default: process.env.STRAPI_TRANSFER_URL,
+                  validate(input) {
+                    try {
+                      const url = new URL(input);
+                      if (!['http:', 'https:'].includes(url.protocol)) {
+                        return 'URL must use http: or https: protocol';
+                      }
+                      return true;
+                    } catch (error) {
+                      return 'Please enter a valid URL (e.g., http://localhost:1337 or https://example.com)';
+                    }
+                  }
+                },
+                {
+                  type: 'password',
+                  name: 'token',
+                  message: 'Enter your transfer token for the remote Strapi destination:',
+                  default: process.env.STRAPI_TRANSFER_TOKEN,
+                  validate(input) {
+                    if (!input?.length) {
+                      return 'Transfer token is required';
+                    }
+                    return true;
+                  }
+                }
+              ]);
+              opts.to = new URL(remoteUrl);
+              opts.toToken = token;
+            }
+          }
+          
+          // Validate that exactly one direction is specified
+          if (opts.from && opts.to) {
             exitWith(
               1,
               'Exactly one remote source (from) or destination (to) option must be provided'
-            )
-        )
+            );
+          }
```

Comment:
Can we keep the error check in its own hook?  basically leave the exact code that was already there, and just put it after your new hook?